REB_firmware/REB_firmware_update/v0/producer_REB_firmware_update.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+from ccsTools import ccsProducer
+from java.lang import Exception
+import Tkinter
+import glob
+import os
+import subprocess
+
+CCS.setThrowExceptions(True);
+
+#Copy firmware_load_scripts files
+instdir = os.getenv('LCATR_INSTALL_AREA').replace('share/','')
+os.system('cp -vp %s/REB_v5/firmware_load_scripts/bin/et_prog_flash.sh .', instdir)
+os.system('cp -vp %s/REB_v5/firmware_load_scripts/bin/flash_load.tcl .', instdir)
+
+#Safety button
+apptxt = "Click here when ready to start firmware update of the REB"
+print apptxt
+top = Tkinter.Tk()
+A = Tkinter.Button(top, text = apptxt, command = top.destroy, bg = "yellow", font = ("Helvetica",24))
+A.pack()
+top.title('Are you ready to start the firmware update?')
+top.mainloop()
+
+#Get list of files in firmware directory
+list_of_firmware_files = glob.glob(os.path.join(instdir,'/REB_v5/targets/REB_v5_top/images/', '*.mcs'))
+
+#Find most recent file
+recent_file = max(list_of_firmware_files, key=os.path.getctime)
```

Comment:
In Step 4, the operator compares the currently installed version as ascertained in Step 3 against the desired one to decide whether an upgrade is needed.  How does the operator know the version of desired firmware in order to make that decision?   If the desired version were available programmatically somehow (I don't know how offhand), then the code could retrieve the desired version from that same source.